invenio_accounts/datastore.py

```diff
@@ -19,14 +23,46 @@
 class SessionAwareSQLAlchemyUserDatastore(SQLAlchemyUserDatastore):
     """Datastore which deletes active session when a user is deactivated."""
 
+    def verify_user(self, user):
+        """Verify a user."""
+        now = datetime.utcnow()
+        user.blocked_at = None
+        user.verified_at = now
+        if user.confirmed_at is None:
+            user.confirmed_at = now
+        if not user.active:
+            user.active = True
+        return True
+
+    def block_user(self, user):
+        """Verify a user."""
+        now = datetime.utcnow()
+        user.blocked_at = now
+        user.verified_at = None
+        if user.active:
+            user.active = False
+        delete_user_sessions(user)
+        return True
+
+    def activate_user(self, user):
+        """Activate a unconfirmed/deactivated/blocked user."""
+        res = super().activate_user(user)
+        user.blocked_at = None
+        if user.confirmed_at is None:
+            user.confirmed_at = datetime.utcnow()
+            user_confirmed.send(current_app._get_current_object(), user=user)
```

Comment:
Needs to send signal to run the code which takes care of registering the domain.